server/src/main/java/org/elasticsearch/common/settings/Settings.java

```diff
@@ -133,8 +133,7 @@ private Settings(Map<String, Object> settings, SecureSettings secureSettings) {
     /**
      * Retrieve the secure settings in these settings.
      */
-    SecureSettings getSecureSettings() {
-        // pkg private so it can only be accessed by local subclasses of SecureSetting
+    public SecureSettings getSecureSettings() {
```

Comment:
I'm not super happy about this, but I couldn't find another way to provide the secure settings during the desired nodes validation. I'm happy to hear your feedback.